vscode/src/chat/ChatPanelProvider.ts

```diff
@@ -0,0 +1,459 @@
+            return
+        }
+
+        const edit = new vscode.WorkspaceEdit()
+        // trimEnd() to remove new line added by Cody
+        edit.insert(editor.document.uri, selectionRange.start, text + '\n')
+        await vscode.workspace.applyEdit(edit)
+
+        // Log insert event
+        const op = 'insert'
+        const eventName = op + 'Button'
+        this.editor.controllers.inline?.setLastCopiedCode(text, eventName, source)
+    }
+
+    /**
+     * Handles insert event to insert text from code block to new file
+     */
+    private async handleSaveToNewFile(text: string, source?: string): Promise<void> {
+        // Log insert event
+        const op = 'save'
+        const eventName = op + 'Button'
+        this.editor.controllers.inline?.setLastCopiedCode(text, eventName, source)
+
+        await this.editor.createWorkspaceFile(text)
+    }
+
+    /**
+     * Handles copying code and detecting a paste event.
+     * @param text - The text from code block when copy event is triggered
+     * @param eventType - Either 'Button' or 'Keydown'
+     */
+    private async handleCopiedCode(text: string, eventType: 'Button' | 'Keydown', source?: string): Promise<void> {
+        // If it's a Button event, then the text is already passed in from the whole code block
+        const copiedCode = eventType === 'Button' ? text : await vscode.env.clipboard.readText()
+        const eventName = eventType === 'Button' ? 'copyButton' : 'keyDown:Copy'
+        // Send to Inline Controller for tracking
+        if (copiedCode) {
+            this.editor.controllers.inline?.setLastCopiedCode(copiedCode, eventName, source)
+        }
+    }
+
+    protected handleCodyCommands(prompts: [string, CodyPrompt][]): void {
+        void this.webview?.postMessage({
+            type: 'custom-prompts',
+            prompts,
+        })
+    }
+
+    /**
+     *
+     * @param notice Triggers displaying a notice.
+     * @param notice.key The key of the notice to display.
+     */
+    public triggerNotice(notice: { key: string }): void {
+        // They may not have chat open, and given the current notices are
+        // designed to be triggered once only during onboarding, we open the
+        // chat view. If we have other notices and this feels too aggressive, we
+        // can make it be conditional on the type of notice being triggered.
+        void vscode.commands.executeCommand('cody.chat.focus', {
+            // Notices are not meant to steal focus from the editor
+            preserveFocus: true,
+        })
+        void this.webview?.postMessage({
+            type: 'notice',
+            notice,
+        })
+    }
+
+    /**
+     * Set webview view
+     */
+    public async setWebviewView(view: View): Promise<void> {
+        if (view !== 'chat') {
+            return
+        }
+        this.webviewPanel?.reveal(vscode.ViewColumn.Beside, !!view)
+        return Promise.resolve()
+    }
```

Comment:
`async` is not required here anymore:

```suggestion
    public setWebviewView(view: View): void {
        if (view !== 'chat') {
            return
        }
        this.webviewPanel?.reveal(vscode.ViewColumn.Beside, !!view)
    }
```